aries-site/src/pages/templates/forms.mdx

```diff
@@ -23,29 +23,67 @@ import { Example } from '../../layouts';
-- Fill in default values for as many fields as possible
-- Reveal fields progressively so the user is not overwhelmed
-- Use clear, concise labels for each field
-- Use placeholder text as an added visual indication of expected input value format
+#### Do's
+
+- Ask only for the required information. Keep the fields to a minimum to allow the user to finish their task as quickly as possible.
+- User-entered input needs to be encoded to prevent [SQL injection](https://www.w3schools.com/sql/sql_injection.asp) and [XSS attacks](https://en.wikipedia.org/wiki/Cross-site_scripting).
+- Use asterisks for required fields when the form has more than two required fields and/or has optional fields. [See HPE UX research](https://www.figma.com/file/3fkwBelW5UsCbfwdDCJkT8/HPE-Form-Templates?node-id=4988%3A3653).  
+- Asterisks for required fields should be the same color as label field 
+- If a field is optional, write "Optional" in italics next to the form field label
+- Wrap all input elements within a FormField `<FormField label="Username"><TextInput /></FormField>`  
+- Reveal fields progressively so the user is not overwhelmed 
+- Write in sentence case (capitalize the first letter and all following letters should be lowercase)  
+- Fill in default values for as many fields as possible 
+- Use clear, concise labels for each field 
+- Write helpful placeholder text inside the fields 
+- If absolutely necessary, include helpful text below the field, such as [password requirements](#signup) or [additional resources](#customize) on a field's context
+- List the most common field requirements below the relevant field 
+- Place the submit, reset and cancel buttons at the bottom left of the form. [See responsive button behavior](https://design-system.hpe.design/components/button?q=button#responsiveness). 
+- Use specific labels on buttons that tell the user what action to expect 
+- Provide feedback when the form is submitted successfully (e.g. notification, success page, green check mark, etc.) 
+- [In the case of an error](#requiredoptional), red highlight the field where the error occurred 
+- List below the button all relevant errors and the corrective action the user must take 
+- Allow user to continue navigating through the application immediately
+
+#### Don'ts
+
+- Have more than two optional fields  
+- Repeat the field label to use as placeholder text 
+- Use asterisks for sign up and sign in forms 
+- Red asterisks stresses the user and actually decrease success in filling out the form 
+- Use generic labels on buttons, such as “Submit” 
+- Restrict resource naming. Allow spaces, punctuation and unicode characters (ex. Björk Guðmundsdóttir, 联想, John Q. Public, etc.)
+- No need to list requirements up front that the user is unlikely to commit (ex. double hyphens, starting with a space, etc.) 
 
-### Required vs. optional fields
 
-- When possible, restrict forms only to fields that are required in order to keep them concise.
+### Single field forms
```

Comment:
Is this an "exception"?

Reply:
I'm trying to make it easy for users to just go to one section where we list all of the guidance in ideal circumstances. The opposite of this is forcing the user to jump around to different sections to have to figure out what the "do's" are.